homeassistant/config.py

```diff
@@ -116,6 +117,19 @@
 """
 
 
+@dataclass
+class ConfigExceptionInfo:
+    """Configuration exception info class."""
+
+    ex: Exception
+    translation_key: str
+    platform_name: str
+    config: ConfigType
+    log_message: str | None = None
+    integration_link: str | None = None
+    log_exception: bool = False
```

Comment:
This is not a good name, because this class is all about logging after an exception happened.
Maybe call it `show_stack_trace`, or `unexpected` instead?